packages/blaze/lookup.js

```diff
@@ -84,6 +84,25 @@ var wrapHelper = function (f, templateFunc) {
   };
 };
 
+function _lexicalKeepGoing(currentView) {
+  if (!currentView.parentView) {
+    return undefined;
+  }
+  if (!currentView.__startsNewLexicalScope) {
+    return currentView.parentView;
+  }
+  if (currentView.parentView.__childDoesntStartNewLexicalScope) {
+    return currentView.parentView;
+  }
+  
+  // in the case of {{> Template.contentBlock data}} the contentBlock loses the lexical scope of it's parent, wheras {{> Template.contentBlock}} it does not
+  // this is because a #with sits between the include InOuterTemplateScope
+  if (currentView.parentView.name === "with" && currentView.parentView.parentView && currentView.parentView.parentView.__childDoesntStartNewLexicalScope) {
```

Comment:
I would personally break the lines for better readability but that's optional, since we have no definite linter yet (I think)